src/components/modals/TransferPDNTModal/TransferPDNTModal.tsx

```diff
@@ -0,0 +1,260 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { useArweaveCompositeProvider, useIsMobile } from '../../../hooks';
+import { useGlobalState } from '../../../state/contexts/GlobalState';
+import { useTransactionState } from '../../../state/contexts/TransactionState';
+import {
+  ArweaveTransactionID,
+  INTERACTION_TYPES,
+  PDNTContractJSON,
+  VALIDATION_INPUT_TYPES,
+} from '../../../types';
+import {
+  getAssociatedNames,
+  mapTransactionDataKeyToPayload,
+} from '../../../utils';
+import { AlertTriangleIcon, CloseIcon } from '../../icons';
+import CopyTextButton from '../../inputs/buttons/CopyTextButton/CopyTextButton';
+import ValidationInput from '../../inputs/text/ValidationInput/ValidationInput';
+import { Loader } from '../../layout';
+
+function TransferPDNTModal({
+  pdntId,
+  showModal,
+}: {
+  pdntId: ArweaveTransactionID; // contract ID if asset type is a contract interaction
+  showModal: () => void;
```

Comment:
nit: this could potentially be a `transfer` route and page, but will leave for now